backend/task/data_from_fire.py

```diff
@@ -0,0 +1,205 @@
+"""
+@author: Scarlett Zhang
+This file has 1 class:
+DataFromFire: data pipeline for fire data.
+"""
+import os
+import logging
+from datetime import datetime
+import datetime
+import rootpath
+import psycopg2
+from paths import FIRE_DATA_DIR
+from typing import Tuple, List
+import time
+
+rootpath.append()
+from backend.task.runnable import Runnable
+from backend.data_preparation.crawler.fire_crawler import FireCrawler
+from backend.data_preparation.extractor.fire_extractor import FireExtractor, IncompleteShapefileError
+from backend.data_preparation.dumper.fire_dumper import FireDumper
+
+logger = logging.getLogger('TaskManager')
+
+
+class DataFromFire(Runnable):
+    """
+    Run once per day/week/, not any time
+    """
+    def __init__(self):
+        """
+        Initialize runnable object
+        """
+        # crawler object initialization: can add more state names to crawl
+        self.crawler = FireCrawler(["California"])
+        # extractor object initialization
+        self.extractor = FireExtractor()
+        # dumper object initialization
+        self.dumper = FireDumper()
+        self.start_time = 2015
+
+    @staticmethod
+    def create_temporary_data_path():
+        """
+        Checks if the temporary directory (FIRE_DATA_DIR) exists, if yes, do nothing
+        if not, create the temporary directory.
+        """
+        # check if the FIRE_DATA_DIR exists,
+        # this is the temporary directory to store fire shapefile downloaded from rmgsc
+        if not os.path.isdir(FIRE_DATA_DIR):
+            # if the temporary directory does not exist, create it
+            logger.info(f"No fire data folder detected. Creating a new one at: {FIRE_DATA_DIR}")
+            os.makedirs(FIRE_DATA_DIR)
+        else:
+            logger.info(f"Temp fire data folder detected, path: {FIRE_DATA_DIR}")
+
+    @staticmethod
+    def handle_fire_page_tuples(fire_page_tuple: Tuple[str], fire_id) -> Tuple:
+        """
+        Takes a tuple of information about a fire webpage and returns a tuple of unified information
+        :param fire_page_tuple: Tuple
+        :param fire_id: int
+        :return: Tuple
+        """
+        if len(fire_page_tuple) == 4:
+            # if it is a TYPE 2 record
+            fire_id = fire_page_tuple[0]
+            year = fire_page_tuple[1]
+            state = fire_page_tuple[2]
+            urlname = fire_page_tuple[3]
+            logger.info(f"Updating recent fire: old fire id:{fire_id}, year: {year}, state:{state}, urlname:{urlname}.")
+        else:
+            # if it is a TYPE 1 record
+            year = fire_page_tuple[0]
+            state = fire_page_tuple[1]
+            urlname = fire_page_tuple[2]
+            logger.info(f"Updating new fire: fire id:{fire_id}, year: {year}, state:{state}, urlname:{urlname}.")
```

Comment:
this log msg is not accurate. you are not updating fire information yet. Unless by updating, you did not mean database sql UPDATE